### PR TITLE
Add db init script for mongodb to create required collections

### PR DESCRIPTION
**Pull request by jakecomp**
# Mongodb Init Script 
In this PR we created a simple initialization script which creates the initial schedule database, and all the required collections including: 
1. users 
2. classrooms 
3. courses
4. schedules 

## Files of interest (under mongodb directory) 
1. Dockerfile 
2. mongo-init.js 

## Test / Demo instructions 

1. Run docker-compose: 
`docker-compose up -d` 

2. Verify the mongo container is up and in a healthy state: 
`docker ps -a` 

3. Exec into the container: 
`docker exec -it backend_mongo_db_1 bash` then login to mongo `mongo -u <username> -p <password> ` 

4. Verify you can see schedule database: 
`show databases`, and then run `use schedule_db` 

5. Finally verify all collections exist: 
`show collections`

### CODE REVIEW

File: mongodb/mongo-init.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+db = db.getSiblingDB("admin") 
+db.auth("admin","admin")
```
Comment (jberryman566) [inline mongodb/mongo-init.js]: Some security issues come to mind looking at this, we should look into best practices for passing credentials in without hard coding them.

Comment (jakecomp) [reply mongodb/mongo-init.js]: I agree. I will look into more secure ways of managing our credentials

**Comment by jakecomp** [comment]
So now I have removed all hard coded credentials using docker secrets. Currently credentials are stored locally in .txt files but for security purposes these files shouldn't be pushed up. I'm thinking we can store the credentials locally in .env files since this is slightly more secure on our local machines. Either way no credentials are ever stored on the repo. Would you like me to go ahead with .env files for the docker secrets?

**Comment by AnthonyRodway** [comment]
> So now I have removed all hard coded credentials using docker secrets. Currently credentials are stored locally in .txt files but for security purposes these files shouldn't be pushed up. I'm thinking we can store the credentials locally in .env files since this is slightly more secure on our local machines. Either way no credentials are ever stored on the repo. Would you like me to go ahead with .env files for the docker secrets?

I think that could work well, we would just need to have separate local files that don't get pushed into the repo.

**Comment by AnthonyRodway** [comment]
When running the commands, the docker-compose creates the container perfectly but automatically exits mongo_db-1. Is this a me problem or are others also experiencing it?

**Comment by jakecomp** [comment]
> > So now I have removed all hard coded credentials using docker secrets. Currently credentials are stored locally in .txt files but for security purposes these files shouldn't be pushed up. I'm thinking we can store the credentials locally in .env files since this is slightly more secure on our local machines. Either way no credentials are ever stored on the repo. Would you like me to go ahead with .env files for the docker secrets?
> 
> I think that could work well, we would just need to have separate local files that don't get pushed into the repo.



> When running the commands, the docker-compose creates the container perfectly but automatically exits mongo_db-1. Is this a me problem or are others also experiencing it?

Thats odd... I haven't experienced that. Do you mind showing the output of `docker logs mongo_db-1` ?

**Comment by AnthonyRodway** [comment]
> Thats odd... I haven't experienced that. Do you mind showing the output of `docker logs mongo_db-1` ?

It immediately exits even when I try to run manually in docker desktop so the logs are completely blank

**Comment by AnthonyRodway** [comment]
> > Thats odd... I haven't experienced that. Do you mind showing the output of `docker logs mongo_db-1` ?
> 
> It immediately exits even when I try to run manually in docker desktop so the logs are completely blank

Nevermind, it seems to have fixed itself by doing the good old delete everything and redo the docker-compose 😅

**Comment by AnthonyRodway** [comment]
Notes for others (I am on windows using vscode and docker desktop):
- If your docker compose is creating containers but your mongo_db-1 container is exiting immediately, it maye be creating directories for the .mongo_pw and .mongo_user env files. If that happens, then just manually create them as files instead putting in any username and password that you want.

- To exec into the container, if you are having issues you may need to use 
`docker exec -it backend-mongo_db-1 bash` or something similar to what your machine created the containers as

**Review by AnthonyRodway — APPROVED** [review]
Init script is working and successfully created the db with required collections.

**Pull request merged**